static/app/views/issueList/customViewsHeader.tsx

```diff
@@ -0,0 +1,436 @@
+  }, []);
+
+  // Update local tabs when new views are received from mutation request
+  useEffect(() => {
+    setDraggableTabs(
+      draggableTabs.map(tab => {
+        if (!tab.id) {
+          views.forEach(view => {
+            if (
+              tab.query === view.query &&
+              tab.querySort === view.querySort &&
+              tab.label === view.name
+            ) {
+              tab.id = view.id;
+              tab.to = normalizeUrl({
+                query: {
+                  ...queryParams,
+                  query: view.query,
+                  sort: view.querySort,
+                  viewId: view.id,
+                },
+                pathname: `/organizations/${organization.slug}/issues`,
+              });
+            }
+          });
+        }
+        return tab;
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [views]);
+
+  // Set parent component's bottem border to dashed if temp view is selected
+  useEffect(() => {
+    if (tempTab && selectedTabKey === 'temporary-tab') {
+      setBorderStyle('dashed');
+    } else {
+      setBorderStyle('solid');
+    }
+  }, [tempTab, selectedTabKey, setBorderStyle]);
+
+  // Updates the tab's hasSavedChanges state
+  useEffect(() => {
+    const currentTab = draggableTabs?.find(tab => tab.key === selectedTabKey);
+    if (
+      currentTab &&
+      (query !== currentTab.query || sortParam !== currentTab.querySort)
+    ) {
+      // console.log(currentTab.query, query, currentTab.querySort, sortParam);
+      setDraggableTabs(
+        draggableTabs?.map(tab => {
+          if (tab.key === selectedTabKey) {
+            tab.unsavedChanges = [query, sortParam];
+            tab.to = normalizeUrl({
+              query: {
+                ...queryParams,
+                query,
+                sort: sortParam,
+                ...(tab.id ? {viewId: tab.id} : {}),
+              },
+              pathname: `/organizations/${organization.slug}/issues/`,
+            });
+          }
+          return tab;
+        })
+      );
+    } else if (
+      currentTab &&
+      query === currentTab.query &&
+      sortParam === currentTab.querySort
+    ) {
+      setDraggableTabs(
+        draggableTabs?.map(tab => {
+          if (tab.key === selectedTabKey) {
+            tab.unsavedChanges = undefined;
+          }
+          return tab;
+        })
+      );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query, sortParam]);
+
+  // Loads query counts when they are available
+  useEffect(() => {
+    setDraggableTabs(
+      draggableTabs?.map(tab => {
+        if (tab.query && queryCounts[tab.query]) {
+          tab.queryCount = queryCounts[tab.query]?.count ?? 0; // TODO: Confirm null = 0 is correct
+        }
+        return tab;
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [queryCounts]);
+
+  const onAddView = () => {
```

Comment:
The state management of these tabs is tangled between CustomViewsHeader and DraggableTabBar right now, though I'd ideally like to move the tab state management to DraggableTabBar in the future and just have CustomViewsHeader worry about the debouncing and which queries/sort to navigate to. This is easy enough to do for `onDelete`, for example, but `onAddView` and `onDiscard` require some more information about the query being displayed, so I have left the handlers for this in CustomViewsHeader for now. 

I'd like to generalize DraggableTabBar as much as possible in general, but in the interest of getting custom views out for internal testing as fast as possible, I'll leave this up to future me to do in future PRs, or while this PR is getting reviewed.